tests/backup/backup_helper.go

```diff
@@ -5559,3 +5680,197 @@ func UpgradeKubevirt(versionToUpgrade string, workloadUpgrade bool) error {
+	}
+
+	// Removing all old stork pods
+	for _, pod := range storkOldPods.Items {
+		err = core.Instance().DeletePod(pod.Name, pod.Namespace, false)
+		if err != nil {
+			log.Warnf("Unable to delete [%v]. Error - [%s]", pod.Name, err.Error())
+		} else {
+			log.Infof("Old stork pod deleted [%v]", pod.Name)
+		}
+	}
+
+	time.Sleep(1 * time.Minute)
+
+	err = apps.Instance().ValidateDeployment(updatedStorkDeployment, storkPodReadyTimeout, podReadyRetryTime)
+	if err != nil {
+		return nil, err
+	}
+
+	return stc, nil
+}
+
+// getCurrentAdminNamespace returns the value of current admin namespace set
+func getCurrentAdminNamespace() (string, error) {
+	isOpBased, _ := Inst().V.IsOperatorBasedInstall()
+	if isOpBased {
+		stc, err := Inst().V.GetDriver()
+		if err != nil {
+			return "", err
+		}
+		if adminNamespace, ok := stc.Spec.Stork.Args["admin-namespace"]; ok {
+			log.Infof("Current admin namespace - [%s]", adminNamespace)
+			return adminNamespace, nil
+		} else {
+			adminNamespace, _ := k8sutils.GetStorkPodNamespace()
+			log.Infof("Current admin namespace - [%s]", adminNamespace)
+			return adminNamespace, nil
+		}
+
+	} else {
+		adminNamespace, err := k8sutils.GetStorkPodNamespace()
+		if err != nil {
+			return "", err
+		}
+		log.Infof("Current admin namespace - [%s]", adminNamespace)
+		return adminNamespace, nil
+	}
+}
+
+// Validates Backup CRs created
+func ValidateBackupCRs(backupName string, clusterName string, orgID string, uid string, ctx context.Context,
+	backupNameSpaces []string) error {
+
+	currentAdminNamespace, _ := getCurrentAdminNamespace()
+	if len(backupNameSpaces) == 1 {
+		currentAdminNamespace = backupNameSpaces[0]
+	}
+	log.Infof("Current CR Namespace: [%s]", currentAdminNamespace)
+
+	backupDriver := Inst().Backup
+	clusterReq := &api.ClusterInspectRequest{OrgId: orgID, Name: clusterName, IncludeSecrets: true, Uid: uid}
+	clusterResp, err := backupDriver.InspectCluster(ctx, clusterReq)
+	if err != nil {
+		return err
+	}
+	clusterObj := clusterResp.GetCluster()
+
+	validateBackupCRInNameSpace := func() (interface{}, bool, error) {
+		allBackupCrs, err := backupDriver.ListBackupCRs(ctx, currentAdminNamespace, clusterObj, orgID)
+		if err != nil {
+			return false, true, err
+		}
+		log.InfoD("All backup CRs in [%s] are [%v]", currentAdminNamespace, allBackupCrs)
+
+		for _, eachCR := range allBackupCrs {
+			if strings.Contains(eachCR, backupName) {
+				log.Infof("Backup CR found for [%s] under [%s] namespace", backupName, currentAdminNamespace)
+				return false, false, nil
+			}
+		}
+		return false, true, fmt.Errorf("Unable to find CR for [%s] under [%s] namespace", backupName, currentAdminNamespace)
+	}
+
+	_, err = task.DoRetryWithTimeout(validateBackupCRInNameSpace, 5*time.Minute, 500*time.Millisecond)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Validates Restore CRs created
+func ValidateRestoreCRs(restoreName string, clusterName string, orgID string, uid string, ctx context.Context,
```

Comment:
we should explore WatchPods

Reply:
Will take up this validation as part of separate PR